services/frontend-service/src/ui/components/ServiceLane/ServiceLane.tsx

```diff
@@ -104,18 +105,48 @@ const deriveUndeployMessage = (undeploySummary: UndeploySummary | undefined): st
     }
 };
 
-export const ServiceLane: React.FC<{ application: OverviewApplication; hideMinors: boolean }> = (props) => {
+export const ServiceLane: React.FC<{
+    application: OverviewApplication;
+    hideMinors: boolean;
+}> = (props) => {
     const { application, hideMinors } = props;
     const { authHeader } = useAzureAuthSub((auth) => auth);
 
     const appDetails = useAppDetailsForApp(application.name);
+    const componentRef: React.MutableRefObject<any> = React.useRef();
+
     React.useEffect(() => {
-        getAppDetails(application.name, authHeader);
-    }, [application, authHeader]);
+        if (componentRef.current !== null) {
+            const rect = componentRef.current.getBoundingClientRect();
+            if (rect.top >= 0 && rect.bottom <= window.innerHeight) {
+                if (appDetails.appDetailState === AppDetailsState.NOTREQUESTED) {
+                    getAppDetails(application.name, authHeader);
+                }
+            }
+        }
+        const handleScroll = (): void => {
+            if (componentRef.current !== null) {
+                const rect = componentRef.current.getBoundingClientRect();
+                if (rect.top >= 0 && rect.bottom <= window.innerHeight) {
+                    if (appDetails.appDetailState === AppDetailsState.NOTREQUESTED) {
+                        getAppDetails(application.name, authHeader);
+                    }
+                }
+            }
+        };
+        if (document.getElementsByClassName('mdc-drawer-app-content').length !== 0) {
+            document.getElementsByClassName('mdc-drawer-app-content')[0].addEventListener('scroll', handleScroll);
```

Comment:
cant' we use `componentRef.current.addEventlistener` here? 
Running getElementsByClassName sounds like an unneccessary indirection.

Reply:
componentRef here represents the reference to the service lane div. This is not the scrollable element, the scrollable element is the element with the  'mdc-drawer-app-content' class. There is no easy way of passing a reference to this element to the service lane